base/pmr/arena.h

```diff
@@ -7,15 +7,25 @@
 #include <cassert>
 #include <cstddef>
 #include <cstdint>
+
+#ifdef __clang__
```

Comment:
We should `#include <version>` and use
```suggestion
#ifndef __cpp_lib_memory_resource
```
instead, they added it in libc++16 and we can use clang+libstdc++ combination as well.

Reply:
but there are a couple other places where we do this as well so maybe another PR